src/treetensornetworks/projttns/projttn_apply.jl

```diff
@@ -1,57 +1,114 @@
 struct ProjTTNApply{V} <: AbstractProjTTN{V}
   pos::Union{Vector{<:V},NamedEdge{V}}
-  psi0::TTN{V}
-  H::TTN{V}
+  init_state::AbstractTTN{V}
+  operator::AbstractTTN{V}
   environments::Dictionary{NamedEdge{V},ITensor}
 end
 
-function ProjTTNApply(psi0::TTN, H::TTN)
-  return ProjTTNApply(vertextype(H)[], psi0, H, Dictionary{edgetype(H),ITensor}())
+init_state(p::ProjTTNApply) = p.init_state
+
+function ProjTTNApply(init_state::AbstractTTN, operator::AbstractTTN)
+  return ProjTTNApply(
+    vertextype(operator)[], init_state, operator, Dictionary{edgetype(operator),ITensor}()
+  )
 end
 
 function copy(P::ProjTTNApply)
-  return ProjTTNApply(P.pos, copy(P.psi0), copy(P.H), copy(P.environments))
+  return ProjTTNApply(
+    pos(P), copy(init_state(P)), copy(operator(P)), copy_keys_values(environments(P))
+  )
 end
 
+function unsafe_copy(P::ProjTTNApply)
```

Comment:
Could you explain this function?

Reply:
This is basically only there for testing purposes, and will become the normal copy function with the fix in Dictionaries.jl